code/modules/cargo/orderconsole.dm

```diff
@@ -164,6 +179,107 @@
 		))
 	return data
 
+/obj/machinery/computer/cargo/proc/add_item(params)
+	if(is_express)
+		return
+	var/id = params["id"]
+	id = text2path(id) || id
+	var/datum/supply_pack/pack = SSshuttle.supply_packs[id]
+	if(!istype(pack))
+		CRASH("Unknown supply pack id given by order console ui. ID: [params["id"]]")
+	if((pack.hidden && !(obj_flags & EMAGGED)) || (pack.contraband && !contraband) || pack.drop_pod_only || (pack.special && !pack.special_enabled))
+		return
+
+	var/name = "*None Provided*"
+	var/rank = "*None Provided*"
+	var/ckey = usr.ckey
+	if(ishuman(usr))
+		var/mob/living/carbon/human/H = usr
+		name = H.get_authentification_name()
+		rank = H.get_assignment(hand_first = TRUE)
+	else if(issilicon(usr))
+		name = usr.real_name
+		rank = "Silicon"
+
+	var/datum/bank_account/account
+	if(self_paid && isliving(usr))
+		var/mob/living/L = usr
+		var/obj/item/card/id/id_card = L.get_idcard(TRUE)
+		if(!istype(id_card))
+			say("No ID card detected.")
+			return
+		if(istype(id_card, /obj/item/card/id/departmental_budget))
+			say("The [src] rejects [id_card].")
+			return
+		account = id_card.registered_account
+		if(!istype(account))
+			say("Invalid bank account.")
+			return
+		var/list/access = id_card.GetAccess()
+		if(pack.access_view && !(pack.access_view in access))
+			say("[id_card] lacks the requisite access for this purchase.")
+			return
+
+	var/reason = ""
+	if(requestonly && !self_paid)
+		reason = tgui_input_text(usr, "Reason", name)
+		if(isnull(reason))
+			return
+
+	if(pack.goody && !self_paid)
+		playsound(src, 'sound/machines/buzz-sigh.ogg', 50, FALSE)
+		say("ERROR: Small crates may only be purchased by private accounts.")
+		return
+
+	var/amount = params["amount"]
+	for(var/count in 1 to amount)
+		var/obj/item/coupon/applied_coupon
+		for(var/i in loaded_coupons)
+			var/obj/item/coupon/coupon_check = i
+			if(pack.type == coupon_check.discounted_pack)
+				say("Coupon found! [round(coupon_check.discount_pct_off * 100)]% off applied!")
+				coupon_check.moveToNullspace()
+				applied_coupon = coupon_check
+				break
+
+		var/datum/supply_order/SO = new(pack, name, rank, ckey, reason, account, null, applied_coupon)
```

Comment:
`order`
Also at calls like these with extensive params - it may be beneficial to explicitly name the params (you wouldn't need to pass in a null either)